fsm/join_add_csync.go

```diff
@@ -121,40 +120,15 @@ func JoinAddCsyncAction(z *music.Zone) bool {
 		}
 
 		log.Printf("%s: Creating CSYNC record sets", z.Name)
-		err, rrs := updater.FetchRRset(signer, z.Name, z.Name, dns.TypeSOA)
-		if err != nil {
-			log.Printf("Error from updater.FetchRRset: %v\n", err)
-		}
 
-		for _, a := range rrs {
-			soa, ok := a.(*dns.SOA)
-			if !ok {
-				continue
-			}
-
-			csync := new(dns.CSYNC)
-			csync.Hdr = dns.RR_Header{
-				Name:   z.Name,
-				Rrtype: dns.TypeCSYNC,
-				Class:  dns.ClassINET,
-				Ttl:    uint32(ttl),
-			}
-			csync.Serial = soa.Serial
-			csync.Flags = 3
-			csync.TypeBitMap = []uint16{dns.TypeA, dns.TypeNS, dns.TypeAAAA}
-
-			updater := music.GetUpdater(signer.Method)
-			if err := updater.Update(signer, z.Name, z.Name,
-				&[][]dns.RR{[]dns.RR{csync}}, nil); err != nil {
-				z.SetStopReason(fmt.Sprintf("Unable to update %s with CSYNC record sets: %s",
-					signer.Name, err))
-				return false
-			}
-			log.Printf("%s: Updated signer %s successfully with CSYNC record sets",
-				z.Name, signer.Name)
+		if err := updater.Update(signer, z.Name, z.Name,
+			&[][]dns.RR{[]dns.RR{z.CSYNC}}, nil); err != nil {
```

Comment:
This looks like a complicated variable, why is it a _pointer_ to a list of lists?

Reply:
This is based on the updater that we inherited, valid point, will look into it.